### DISCUSSION TITLE
MixedBilinearForms on specific domain attributes?

### DISCUSSION THREAD

**Discussion by Heinrich-BR** [mfem/mfem#3996] [General]
Hi, I've noticed that the `MixedBilinearForm` class doesn't have an `AddDomainIntegrator` method which allows you to pass an `Array` of domain attributes to integrate on like the `BilinearForm` class does. Is there a particular reason for this? Having something like this would be helpful for the application I'm currently working on.

**Comment by vladotomov** [comment]
Hi @Heinrich-BR, currently there are 4 types of integrators that can be added to a`MixedBilinearForm`, and for 3 of them the `AddDomainIntegrator` has a version with a marker, for example:
```
   /// Adds a domain integrator. Assumes ownership of @a bfi.
   void AddDomainIntegrator(BilinearFormIntegrator *bfi);

   /// Adds a domain integrator. Assumes ownership of @a bfi.
   void AddDomainIntegrator(BilinearFormIntegrator *bfi,
                            Array<int> &elem_marker);
```
Only the trace face integrators don't have a marker, but this can be added easily, and it will use the internal face attributes from the input mesh. Does this help?

**Reply by v-dobrev** [reply]
Note that interior faces do not have attributes. For interior faces the logic may need to be based on the attributes of the 2 elements adjacent to the face.

**Reply by Heinrich-BR** [reply]
Hi @vladotomov and @v-dobrev . Thank you for your inputs! Yeah my question is really about the domain markers as opposed to face integrators. For some reason I have been running into errors when I try to add a domain integrator with domain markers to a `MixedBilinearForm`. However, as @vladotomov pointed out, this feature is implemented, so I must be doing something wrong. I'll dig more into it and let you know if there are any issues! Thank you for the help
